web/src/lib/managers/cast-manager.svelte.ts

```diff
@@ -0,0 +1,159 @@
+import { GCastDestination } from '$lib/utils/cast/gcast-destination.svelte';
+import { createSession, type SessionCreateResponseDto } from '@immich/sdk';
+import { DateTime, Duration } from 'luxon';
+
+// follows chrome.cast.media.PlayerState
+export enum CastState {
+  IDLE = 'IDLE',
+  PLAYING = 'PLAYING',
+  PAUSED = 'PAUSED',
+  BUFFERING = 'BUFFERING',
+}
+
+export enum CastDestinationType {
+  GCAST = 'GCAST',
+}
+
+export interface ICastDestination {
+  initialize(): Promise<boolean>; // returns if the cast destination can be used
+  type: CastDestinationType; // type of cast destination
+
+  isAvailable: boolean; // can we use the cast destination
+  isConnected: boolean; // is the cast destination actively sharing
+
+  currentTime: number | null; // current seek time the player is at
+  duration: number | null; // duration of media
+
+  receiverName: string | null; // name of the cast destination
+  castState: CastState; // current state of the cast destination
+
+  loadMedia(mediaUrl: string, sessionKey: string, reload: boolean): Promise<void>; // load media to the cast destination
+
+  // remote player controls
+  play(): void;
+  pause(): void;
+  seekTo(time: number): void;
+  disconnect(): void;
+}
+
+class CastManager {
+  private castDestinations = $state<ICastDestination[]>([]);
+  private current = $derived<ICastDestination | null>(this.monitorConnectedDestination());
+
+  availableDestinations = $state<ICastDestination[]>([]);
+  initialized = $state(false);
+
+  isCasting = $derived<boolean>(this.current?.isConnected ?? false);
+  receiverName = $derived<string | null>(this.current?.receiverName ?? null);
+  castState = $derived<CastState | null>(this.current?.castState ?? null);
+  currentTime = $derived<number | null>(this.current?.currentTime ?? null);
+  duration = $derived<number | null>(this.current?.duration ?? null);
+
+  private sessionKey: SessionCreateResponseDto | null = null;
+
+  constructor() {
+    // load each cast destination
```

Comment:
Why are all these comments still here?

Reply:
I felt that they were needed to clarify behavior. I deleted most of the others that were just markers for sections of code. In the future I will try and be more succinct/sparing in my commenting

Reply:
Personally I think almost every one of these comments can be removed and don't add any value to the code. Many just duplicate the meaning implied from the variable and method names